src/backend/distributed/commands/vacuum.c

```diff
@@ -519,13 +571,39 @@ VacuumStmtParams(VacuumStmt *vacstmt)
 		{
 			skip_locked = defGetBoolean(opt);
 		}
+#if PG_VERSION_NUM >= PG_VERSION_16
+		else if (strcmp(opt->defname, "buffer_usage_limit") == 0)
+		{
+			const char *hintmsg;
+			int result;
+			char *vac_buffer_size = defGetString(opt);
+
+			/*
+			 * Check that the specified value is valid and the size falls
+			 * within the hard upper and lower limits if it is not 0.
+			 */
+			if (!parse_int(vac_buffer_size, &result, GUC_UNIT_KB, &hintmsg) ||
```

Comment:
why do we try to do it here? Isn't Postgres already handing it? We execute VACUUM on the coordinator shell table first anyway, and if this is wrong, we should get the error from PG?

Reply:
You are right, I don't need error checks here